### PR TITLE
feat(gateway-protocol): define an openapi specification document

### LINKED ISSUE #16163: Provide an OpenAPI specification in the `gateway-protocol` module
### Description

An OpenAPI specification document is provided in the Zeebe `gateway-protocol` module.

```[tasklist]
### Tasks
- [x] An OpenAPI specification document for the Zeebe REST API is created in the Zeebe `gateway-protocol` module.
- [x] The OpenAPI specification is defined in the YAML format.
- [x] The specification uses OpenAPI version `3.0.2`
- [x] The Zeebe REST API is versioned
- [x] Initial REST server variables are defined (`host`, `port`, `schema`)
```

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Define an OpenAPI specification document in the gateway-protocol module. This document will be used to define the REST API endpoints for Zeebe. It will also be used to generate the appropriate model classes that will be used in the Zeebe REST API controllers.

The OpenAPI specification document currently defines only the OpenAPI spec version, as well as the `info` and `servers` sections. The `info` section provides details on the purpose of the document, the version of the document, license and external documentation. The `servers` section defines the URL format.
## Related issues

closes #16163 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+info:
+  title: Zeebe REST API
+  version: "0.1"
+  description: API for communicating with the Zeebe cluster.
+  license:
+    name: Zeebe Community License Version 1.1
+    url: https://github.com/camunda/zeebe/blob/main/licenses/ZEEBE-COMMUNITY-LICENSE-1.1.txt
+  externalDocs:
+    description: Find out more
+    url: https://docs.camunda.io/docs/apis-tools/zeebe-api-rest/overview/
```
Comment (npepinpe) [inline gateway-protocol/src/main/proto/rest-api.yaml:11]: 💭 i guess this will change?

**Review by npepinpe — APPROVED** [review]

File: gateway-protocol/src/main/proto/rest-api.yaml (hunk 1 above)
Comment (npepinpe) [inline gateway-protocol/src/main/proto/rest-api.yaml:11]: :x: This is invalid, it's a root property, and not a member of `info`

```suggestion
externalDocs:
  description: Find out more
  url: https://docs.camunda.io/docs/apis-tools/zeebe-api-rest/overview/
```

File: gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+openapi: "3.0.2"
```
Comment (npepinpe) [inline gateway-protocol/src/main/proto/rest-api.yaml:1]:
```suggestion
openapi: "3.1.0"
```

We can use the latest version of OpenAPI released in 2021

**Review by npepinpe — CHANGES_REQUESTED** [review]

File: gateway-protocol/src/main/proto/rest-api.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      host:
+        default: localhost
+        description: The hostname of a Zeebe Gateway
+      port:
+        default: "8080"
+        description: The port of the Zeebe REST API server
+      schema:
+        default: http
+        description: The schema of the Zeebe REST API server
+
```
Comment (npepinpe) [inline gateway-protocol/src/main/proto/rest-api.yaml:25]: :x: Missing the required `paths` field (can be left empty for now)

File: gateway-protocol/src/main/proto/rest-api.yaml (hunk 1 above)
Comment (koevskinikola) [reply gateway-protocol/src/main/proto/rest-api.yaml:11]: Yup, it's just a dummy URL for now

**Comment by koevskinikola** [comment]
@npepinpe I implemented you feedback with [a7eec6f](https://github.com/camunda/zeebe/pull/16194/commits/a7eec6f17a98bea331ed9a72aa4d2e0bea5f75c6)

**Review by npepinpe — APPROVED** [review]

**Comment by npepinpe** [comment]
should we merge? :eyes:

**Pull request merged**